entities/StandardGameState.cs

```diff
@@ -0,0 +1,313 @@
+using System;
+using System.Collections.Generic;
+
+
+namespace LightChess
+{
+    /*
+    This class represents the game state. Game state can be uniquely identified by FEN notation.
+    In addition, the + operator is overloaded to get the next game state after move.
+
+    Please refer to FEN notation (https://en.wikipedia.org/wiki/Forsyth%E2%80%93Edwards_Notation). 
+
+    TODO: pawn promotion is not implemented
+    */
+
+    public class StandardGameState: GameState
+    {
+    
+        public StandardBoard Board;
+        public Color ActiveColor;
+        public Color EnemyColor;
+        public List<Castle> AvaialbleCastles;
+        public Cell? EnPassantCell;
+        public int HalfmoveNumber;
+        public int FullmoveNumber;
+
+        public StandardGameState(
+            StandardBoard board,
+            Color color,
+            List<Castle> avaialbleCastles,
+            Cell? enPassantCell,
+            int halfmoveNumber,
+            int fullmoveNumber
+        )
+        {
```

Comment:
I prefer having an opening curly bracket on a previous line, but it is up to us to decide which style guide we follow.